packages/openactive-integration-tests/test/helpers/feature-helper.js

```diff
@@ -169,6 +185,50 @@ class FeatureHelper {
     const bookingFlowsSingleSelection = (SINGLE_FLOW_PATH_MODE || '').split('/')[0];
     const opportunityTypesSingleSelection = (SINGLE_FLOW_PATH_MODE || '').split('/')[1];
 
+    /**
+     * Set up a test which checks that the Test Interface Actions used so far
+     * exactly match those specified in `.testInterfaceActions`.
+     *
+     * To be used at the end of a run of tests (e.g. the end of the
+     * ScheduledSession/ApprovalFlow tests)
+     *
+     * @param {DescribeFeatureRecord} describeFeatureRecord
+     * @param {BookingFlow} [bookingFlow]
+     */
+    const itAssertTestInterfaceActionsUsedAsSpecified = (describeFeatureRecord, bookingFlow) => {
+      const testInterfaceActions = configuration.testInterfaceActions ?? [];
+      /* This constant is also used in report generation. The results from this
+      test will only be shown in report generation if this test fails and all
+      other tests pass. This is to reduce user confusion (as this test is meant
+      to be seen by maintainers only). If other tests fail, then this test may
+      be expected to fail, because the runtime behaviour changes on failure */
+      describe(FEATURE_DESCRIPTION_ASSERTIONS_META_TEST_NAME, () => {
+        it('Feature Description `.testInterfaceActions` should match the Test Interface Actions that were used in the test', () => {
+          const expectedUsedTestInterfaceActions = [...testInterfaceActions].sort();
+          const usedTestInterfaceActions = (() => {
+            const allUsedTestInterfaceActions = [...describeFeatureRecord.getUsedTestInterfaceActions()].sort();
+            /* Approval flow tests will often implicitly use
+            test:SellerAcceptOrderProposalSimulateAction as a necessary part of
+            the booking flow. This therefore does not need to be explicitly
+            annotated */
+            if (bookingFlow === 'OpenBookingApprovalFlow'
+              && !expectedUsedTestInterfaceActions.includes('test:SellerAcceptOrderProposalSimulateAction')
+              && allUsedTestInterfaceActions.includes('test:SellerAcceptOrderProposalSimulateAction')
+            ) {
+              return allUsedTestInterfaceActions
+                .filter(x => x !== 'test:SellerAcceptOrderProposalSimulateAction');
+            }
+            return allUsedTestInterfaceActions;
+          })();
+
+          /* JSON.stringify(..) used because otherwise the test output truncates
+          to `expected [Array(2)] to equal [Array(1)]`, which is useless. */
+          expect(JSON.stringify(usedTestInterfaceActions))
```

Comment:
```suggestion
          /* JSON.stringify(..) used because otherwise the test output truncates
          to `expected [Array(2)] to equal [Array(1)]`, which is useless. 
          The arrays are sorted so string comparison is fine. */
          expect(JSON.stringify(usedTestInterfaceActions))
```
My comment might not be needed 🤷🏽‍♂️